pkg/controller/controller_binding.go

```diff
@@ -87,11 +89,37 @@ func isServiceInstanceCredentialFailed(binding *v1alpha1.ServiceInstanceCredenti
 	return false
 }
 
+func (c *controller) setOrphanMitigation(binding *v1alpha1.ServiceInstanceCredential, toUpdate *v1alpha1.ServiceInstanceCredential, instance *v1alpha1.ServiceInstance, serviceClass *v1alpha1.ServiceClass, brokerName string, errorStr string) error {
+	s := fmt.Sprintf("Setting orphan mitgation for ServiceInstanceCredential \"%s/%s\" for ServiceInstance \"%s/%s\" of ServiceClass %q at ServiceBroker %q, %v",
+		binding.Name,
+		binding.Namespace,
+		instance.Namespace,
+		instance.Name,
+		serviceClass.Name,
+		brokerName,
+		errorStr,
+	)
+	toUpdate.Status.OrphanMitigationInProgress = true
+	toUpdate.Status.OperationStartTime = nil
+	glog.V(5).Info("OrphanMitigationInProgress set to true")
+
+	c.setServiceInstanceCredentialCondition(
+		toUpdate,
+		v1alpha1.ServiceInstanceCredentialConditionReady,
+		v1alpha1.ConditionFalse,
+		"ServiceInstanceCredentialNeedsOrphanMitigation",
+		s,
+	)
+	c.recorder.Event(binding, api.EventTypeWarning, "ServiceInstanceCredentialNeedsOrphanMitigation", s)
+	if _, err := c.updateServiceInstanceCredentialStatus(toUpdate); err != nil {
+		return err
+	}
+	return nil
+}
+
 // an error is returned to indicate that the binding has not been
 // fully processed and should be resubmitted at a later time.
 func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.ServiceInstanceCredential) error {
-	// TODO: this will change once we fully implement orphan mitigation, see:
-	// https://github.com/kubernetes-incubator/service-catalog/issues/988
 	if isServiceInstanceCredentialFailed(binding) && binding.ObjectMeta.DeletionTimestamp == nil {
```

Comment:
The reconciler should perform orphan mitigation even if the binding has the Failed/True condition.